### ISSUE TITLE
.hawkeyeignore not ignoring files in bandit

### ISSUE THREAD

**Issue by nicolastrres** [Stono/hawkeye#36]
Not sure if this is the expected behavior or a bug or I am doing something wrong, but I have been facing some issues while trying to ignore a file.

I am running hawkeye on a python project, and `bandit` is finding some issues on my tests files, I would like to ignore this folder, so I am running hawkeye with the following command:
```
docker run --rm $PWD:/target stono:hawkeye:0.18.0 scan . --exclude ^tests/
```
And I can see the following log when running the tool:
```
[info] Files excluded by exclusion patterns: 14
```

But the `bandit` errors are still showing.

In order to test if I was excluding the tests folder correctly, I have inserted `password='a'` in one of my tests (this would be detected by the `contents` module), and this issue was not show on the errors reported by hawkeye. So, I guess the pattern to be excluded is working well, but this feature is not working for other modules like `bandit`, `brakeman`, etc.

**Comment by Stono** [comment]
@nicolastrres Ahhh this is very interesting actually and not an easy fix at all.  The `--exclude` flag basically excludes directories when the fileManager is collecting a list of files to look up.  So this works for things like [the content scanner](https://github.com/Stono/hawkeye/blob/master/lib/modules/content/index.js#L70)

However other modules which don't simply do test on files, but [shell out to other commands](https://github.com/Stono/hawkeye/blob/master/lib/modules/bandit/index.js#L39) like bandit does.

We would have to have something around [this line](https://github.com/Stono/hawkeye/blob/master/lib/modules/bandit/index.js#L49) which compares the results with those returned from the shelled out command.

I'd like to try and think about a generic way to do this but right now i'm not sure.  Probably because it's saturday morning and I'm knackered.  I'll give it some thought but open to suggestions.

**Comment by Stono** [comment]
@nicolastrres is your project sensitive, could you send it to me as an example that I can test against?

**Comment by nicolastrres** [comment]
Hey, sorry for the delay. I am going to share an example with you as soon as possible.

**Comment by nicolastrres** [comment]
Hello Karl, you can use [this repo](https://github.com/nicolastrres/back-me-up/tree/hawkeye-test) (branch hawkeye-test) to test this. I have added [this line](https://github.com/nicolastrres/back-me-up/blob/hawkeye-test/back_me_up/back_me_up.py#L19) which will be detected by bandit as an issue.

**Comment by elliottmb** [comment]
Any progress on this issue? Running into it as well.

I'm wondering if re-running the exclusion when gathering all the results would handle the cases where the tool does its own file walking.

**Comment by felixhammerl** [comment]
Should be fixed in v1.1.3
